ethcore/src/client/client.rs

```diff
@@ -848,7 +848,7 @@ impl BlockChainClient for Client {
 			difficulty: header.difficulty(),
 			last_hashes: last_hashes,
 			gas_used: U256::zero(),
-			gas_limit: U256::max_value(),
+			gas_limit: header.gas_limit(),
```

Comment:
One of the use cases for gas estimation is to see how much over the gas limit it is. geth sets the hard estimation limit to 50M